### PR TITLE
Feature/us0005 add new post feature and spring-boot-starter-validation

### PR DESCRIPTION
**Pull request by jocabellomeli**

### CODE REVIEW

File: src/main/java/com/mercadolibre/be_java_hisp_w24_g02/repository/implementations/PostRepositoryImpl.java [hunk 1]
```diff
@@ -2,33 +2,52 @@
...
+            return currentPost;
+        }).toList();
+
+        return post;
     }
 
     @Override
     public void delete(Integer id) {
-
+        this.posts = this.posts.stream().filter(post -> Objects.equals(post.getId(), id)).toList();
```
Comment (IColladoMl) [inline src/main/java/com/mercadolibre/be_java_hisp_w24_g02/repository/implementations/PostRepositoryImpl.java:51]: I think maybe here has to be !Objects.equals(post.getId(), id) to create a new list without the post to delete

**Review by stagliaferri — APPROVED** [review]

File: src/main/java/com/mercadolibre/be_java_hisp_w24_g02/exception/GlobalExceptionController.java [hunk 2]
```diff
@@ -14,4 +18,17 @@ public ResponseEntity<ExceptionDAO> notFound(Exception e){
...
+
+    @ExceptionHandler(BadRequestException.class)
+    public ResponseEntity<ExceptionDAO> badRequest(BadRequestException e){
+        ExceptionDAO exceptionDAO = new ExceptionDAO(e.getMessage());
+        return new ResponseEntity<>(exceptionDAO, HttpStatus.BAD_REQUEST);
+    }
+
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleValidationErrors(MethodArgumentNotValidException ex) {
```
Comment (IColladoMl) [inline src/main/java/com/mercadolibre/be_java_hisp_w24_g02/exception/GlobalExceptionController.java:30]: Maybe here can you create a CustomExceptionDTO to not return a plain String

File: src/main/java/com/mercadolibre/be_java_hisp_w24_g02/dao/CreatePostDTO.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.mercadolibre.be_java_hisp_w24_g02.dao;
+
+import jakarta.validation.constraints.NotNull;
+
+public record CreatePostDTO(
+        @NotNull
+        Integer user_id,
```
Comment (IColladoMl) [inline src/main/java/com/mercadolibre/be_java_hisp_w24_g02/dao/CreatePostDTO.java:7]: Try to use cammelCase to name atributtes, you can rename the key of json using @JsonProperty anotation, same to the other atributtes in CreateProductDTO

File: src/main/java/com/mercadolibre/be_java_hisp_w24_g02/service/implementations/PostServiceImpl.java [hunk 4]
```diff
@@ -1,6 +1,48 @@
...
 
+@Service
 public class PostServiceImpl implements IPostService {
+    @Autowired
+    private IPostRepository postRepository;
+    @Autowired
+    private IUserRepository userRepository;
+
+    @Override
+    public Boolean createProductPost(CreatePostDTO createPostDTO) {
```
Comment (IColladoMl) [inline src/main/java/com/mercadolibre/be_java_hisp_w24_g02/service/implementations/PostServiceImpl.java:22]: maybe this method can be void or return the postCreated

**Review by IColladoMl — CHANGES_REQUESTED** [review]

**Review by IColladoMl — APPROVED** [review]

**Pull request merged**